website/en/docs/react/redux.md

```diff
@@ -127,36 +127,45 @@ function bar(value: boolean): BarAction {
 ##### Typing Redux thunk actions <a class="toc" id="toc-typing-redux-thunk-actions" href="#toc-typing-redux-thunk-actions"></a>
 
 In order to type your Redux [thunk actions](http://redux.js.org/docs/advanced/AsyncActions.html#async-action-creators),
-you'll add types for `ThunkAction` as a function `Dispatch`, and `GetState`. `GetState` is a function that returns an `Object`. `Dispatch` accepts a disjoint union of `Action`, `ThunkAction`, `PromiseAction` and `Array<Action>` and can return `any`.
+you'll add types for `ThunkAction` as a function `Dispatch`, and `GetState`. `GetState` is a function that returns `State`.
+`Dispatch` is an intersection of several dispatch flavors: the plain version, the version that works for thunks,
+and the version that works for Promises. Of course you can define only the ones you need.
 
 ```js
-type Dispatch = (action: Action | ThunkAction | PromiseAction) => any;
+import type { Store } from 'redux';
+
+// plain redux
+type PlainDispatch = (action: Action) => Action;
+
+// redux-thunk
+type ThunkDispatch = <R>(action: ThunkAction<R>) => R;
+type ThunkAction<R> = (dispatch: Dispatch, GetState) => R;
 type GetState = () => State;
-type ThunkAction = (dispatch: Dispatch, getState: GetState) => any;
-type PromiseAction = Promise<Action>;
-```
 
-Then to type a thunk action creator, add a return type of a `ThunkAction` to your action creator.
+// redux-promise-middleware
+type PromiseDispatch = <R>(action: PromiseAction<R>) => Promise<{ value: R, action: Action }>;
+type PromiseAction<R> = { payload: Promise<R> } | { payload: { promise: Promise<R> }};
```

Comment:
I haven't actually tried the typing for promises on a real project but I didn't want to leave it out.